src/vs/workbench/browser/parts/editor/editorGroupView.ts

```diff
@@ -1698,14 +1702,12 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 	layout(width: number, height: number): void {
 		this.dimension = new Dimension(width, height);
 
-		// Ensure editor container gets height as CSS depending on the preferred height of the title control
-		const titleHeight = this.titleDimensions.height;
-		const editorHeight = Math.max(0, height - titleHeight);
-		this.editorContainer.style.height = `${editorHeight}px`;
+		// Use the initial height of the title area to layout
+		const initialTitleHeight = this.titleDimensions.height;
+		const titleHeight = this.titleAreaControl.layout(new Dimension(width, initialTitleHeight)) as number;
```

Comment:
Can we not simply pass the full dimensions over that we get above instead of asking the title for its height? Together with the max dimensions for the title widget I thought along the lines of:
* one parameter is the dimension of the entire group and the title widget can use it up to...
* a second parameter defines a certain maximum of size the title widget can have

Based on these 2 parameters the title widget then decides wether it goes multi-line or not.

Reply:
Ok I can try and start working on that, what would you suggest as the maximum size the title widget can have?

Reply:
I would think that the maximum size is:
* `width`: the same as the entire width, aka there is no maximum
* `height`: the entire height minus the minimal height of the editor control

Reply:
Ok I just pushed a very rough implementation that uses a `maxDimensions` parameter to try and remove multi-line tabs. There are problems that arise when testing, presumably due to scheduling the layout on the next frame. I'm sure it needs some fixing so let me know what you think.